libraries/chain/worker_evaluators.cpp

```diff
@@ -480,4 +468,75 @@ namespace golos { namespace chain {
         });
     }
 
+    void worker_payment_approve_evaluator::do_apply(const worker_payment_approve_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_21__1013, "worker_payment_approve_operation");
+
+        auto approver_witness = _db.get_witness(o.approver);
+        GOLOS_CHECK_LOGIC(approver_witness.schedule == witness_object::top19,
+            logic_exception::approver_of_payment_should_be_in_top19_of_witnesses,
+            "Approver of payment should be in Top 19 of witnesses");
+
+        GOLOS_CHECK_LOGIC(o.state != worker_techspec_approve_state::approve,
+            logic_exception::payment_does_not_need_approving_it_can_be_only_disapproved,
+            "Payment does not need approving, it can be only disapproved");
+
+        const auto& wro_post = _db.get_comment(o.worker_result_author, o.worker_result_permlink);
+        const auto& wto = _db.get_worker_result(o.worker_result_author, o.worker_result_permlink);
+
+        const auto& wpo_post = _db.get_comment(wto.worker_proposal_author, wto.worker_proposal_permlink);
+        const auto& wpo = _db.get_worker_proposal(wpo_post.id);
+
+        GOLOS_CHECK_LOGIC(wpo.state == worker_proposal_state::payment,
```

Comment:
It seems strange, that proposal has payment state.
The tspec is received payments, so proposal should has only state - tspec, all other are states of tspec.

Reply:
Lets fix it in another issue, not here.

Reply:
@afalaleev Fixed https://github.com/GolosChain/golos/pull/1137